mars_troughs/accumulation_model.py

```diff
@@ -31,41 +28,203 @@ class InsolationAccumulationModel(AccumulationModel):
         insolations (np.ndarray): value of the solar insolations
     """
 
-    def __init__(self, times: np.ndarray, insolations: np.ndarray):
-        self._ins_times = times
-        self._insolations = insolations
-        self._ins_spline = IUS(self._ins_times, self._insolations)
-        self._ins_spline_integ = self._ins_spline.antiderivative()
-        self._ins2_spline = IUS(self._ins_times, self._insolations ** 2)
-        self._ins2_spline_integ = self.ins2_spline.antiderivative()
+    def __init__(self, times: np.ndarray, insolation: np.ndarray):
+        
+        self.ins_data_spline = IUS(times, insolation)
+        self.iins_data_spline = self.ins_data_spline.antiderivative()
+        self.ins2_data_spline = IUS(times, insolation ** 2)
+        self.iins2_data_spline = self.ins2_data_spline.antiderivative()
+        
+    def get_xt(self, time: np.ndarray, iretreat_model_t_spline: np.ndarray, 
+                                                       angle: float = 2.9):  
+        """
+        Calculates the horizontal distance x (in m) traveled by a point in the
+        center of the high side of the trough. This distance x is a function of 
+        the accumulation rate A and the retreat rate of ice R
+        as in dx/dt=(R(l(t),t)+A(ins(t))cos(theta))/sin(theta). Where theta
+        is the angle of the trough.
+
+        Args:
+            time (np.ndarray): times at which we want the path.
+        Output:
+            horizontal distances (np.ndarray) of the same size as time input.
+        """
+        self.angle=angle           
+        yt = self.get_yt(time)
+        
+        return -self.cot_angle * yt + self.csc_angle * (
+         iretreat_model_t_spline(time) - iretreat_model_t_spline(0))
+    
+    @property
+    def angle(self) -> float:
+        """
+        Slope angle in degrees.
+        """
+        return self._angle * 180.0 / np.pi
+
+    @angle.setter
```

Comment:
Just wanted to explain what this is -- in this class we want to keep track of the `angle`. We don't want to waste compute time and compute the trig functions of the angle over and over again, so we want to comptue them once and save them (the `_csc` and `_cot` attributes).

A `setter` is a kind of method that triggers whenever the identically named property is assigned a value (`angle` in this case). This way, whenever the `angle` is set, the trig functions are computed automatically, so we never risk being in an invalid state where for instance the angle is 90 degrees but the trig function computed cot(92 degrees).